Client/Frontend/Browser/BrowserViewController/BrowserViewController+ProductNotification.swift

```diff
@@ -68,6 +107,27 @@ extension BrowserViewController {
             notifyHttpsUpgrade(theme: Theme.of(selectedTab))
             notificationShown = true
         }
+        
+        // Step 5: Share Brave Benchmark Tiers
+        guard !notificationShown else { return }
+
+        if !Preferences.ProductNotificationBenchmarks.allTiersShown.value {
+            let numOfTrackerAds = BraveGlobalShieldStats.shared.adblock + BraveGlobalShieldStats.shared.trackingProtection
+            let existingTierList = BenchmarkTrackerCountTier.allCases.filter({ numOfTrackerAds > $0.rawValue })
+
+            for tier in existingTierList {
+                if Preferences.ProductNotificationBenchmarks.trackerTierCount.value < numOfTrackerAds {
+                    if let nextTier = tier.nextTier {
+                        Preferences.ProductNotificationBenchmarks.trackerTierCount.value = nextTier.rawValue
+                    } else {
+                        Preferences.ProductNotificationBenchmarks.allTiersShown.value = true
```

Comment:
I wonder if we need this preference, could we just make it a getter that compares current tier with last tier item?
This also would be better in case we want to add new higher tiers in the future

Reply:
Good catch, that makes totally sense, change in https://github.com/brave/brave-ios/pull/3161/commits/63580b10491f1a65bf994163195c0bfeaac23a25